src/workers/transactions/userBoughtCores.ts

```diff
@@ -0,0 +1,68 @@
+import {
+  CioTransactionalMessageTemplateId,
+  sendEmail,
+  updateFlagsStatement,
+} from '../../common';
+import {
+  UserTransaction,
+  UserTransactionProcessor,
+  UserTransactionStatus,
+} from '../../entity/user/UserTransaction';
+import type { TypedWorker } from '../worker';
+
+const purchaseProcessors = [
+  // UserTransactionProcessor.AppleStoreKit,
+  UserTransactionProcessor.Paddle,
+];
+
+export const userBoughtCores: TypedWorker<'api.v1.user-transaction'> = {
+  subscription: 'api.user-bought-cores',
+  handler: async ({ data }, con, logger): Promise<void> => {
+    const transaction = await con.getRepository(UserTransaction).findOneBy({
+      id: data.transaction.id,
+    });
+
+    if (!transaction) {
+      logger.error(
+        { transactionId: data.transaction.id },
+        'Transaction not found',
+      );
+      return;
+    }
+
+    if (!purchaseProcessors.includes(transaction.processor)) {
+      return;
+    }
+
+    if (transaction.status !== UserTransactionStatus.Success) {
+      return;
+    }
+
+    if (transaction.flags.emailSent) {
+      return;
+    }
+
+    const user = await transaction.receiver;
+
+    const coreAmount = Intl.NumberFormat('en-US').format(transaction.value);
+
+    await sendEmail({
+      send_to_unsubscribed: true,
+      transactional_message_id:
+        CioTransactionalMessageTemplateId.UserBoughtCores,
+      message_data: {
+        core_amount: `+${coreAmount}`,
+      },
+      identifiers: {
+        id: user.id,
+      },
+      to: user.email,
+    });
+
+    await con.getRepository(UserTransaction).update(transaction.id, {
+      flags: updateFlagsStatement({
+        emailSent: true,
+      }),
+    });
```

Comment:
I think for safety we will have to do:
- set `emailSent: true` before `sendEmail`
- then await `sendEmail`
- catch and on error set `emailSent: false`

check `dedupedSend` in digest
other

Reply:
Wrapped in try...catch